src/dotnet/APIView/APIViewWeb/Controllers/CopilotCommentsController.cs

```diff
@@ -19,73 +23,126 @@ public CopilotCommentsController(ICopilotCommentsManager copilotManager, ILogger
+            if (id == null || language == null)
             {
-                _logger.LogInformation("Request does not have the required ID field for GET.");
+                _logger.LogInformation("Request does not have the required ID or language fields for GET.");
                 return StatusCode(statusCode: StatusCodes.Status400BadRequest);
             }
 
-            var document = await _copilotManager.GetDocumentAsync(id);
-            if (document != null)
+            try
             {
-                return Ok(document);
-            } else
+                var result = await _copilotManager.GetDocumentAsync(id, language);
+                return Ok(result);
+            }
+            catch (CosmosException)
             {
-                _logger.LogInformation("No document with this id exists in the database.");
+                _logger.LogInformation("Could not find a match for this id and language combination exists in the database.");
                 return StatusCode(statusCode: StatusCodes.Status404NotFound);
             }
+            catch (Exception err)
+            {
+                _logger.LogInformation("Error: unsuccessful GET request. " + err.Message);
+                return StatusCode(statusCode: StatusCodes.Status500InternalServerError);
+            }
+
         }
 
         [HttpDelete]
-        public async Task<ActionResult> DeleteDocument(string id)
+        public async Task<ActionResult> DeleteDocument(string id, string language)
         {
-            if (id == null)
+            if (id == null || language == null)
             {
-                _logger.LogInformation("Request does not have the required ID field for DELETE.");
+                _logger.LogInformation("Request does not have the required ID or language fields for DELETE.");
                 return StatusCode(statusCode: StatusCodes.Status400BadRequest);
             }
 
-            await _copilotManager.DeleteDocumentAsync(User.GetGitHubLogin(), id);
-            return StatusCode(statusCode: StatusCodes.Status204NoContent);
+            try
+            {
+                await _copilotManager.DeleteDocumentAsync(id, language, User.GetGitHubLogin());
+                return StatusCode(statusCode: StatusCodes.Status204NoContent);
+            }
+            catch (CosmosException)
+            {
+                _logger.LogInformation("Could not find a match for this id and language combination exists in the database.");
+                return StatusCode(statusCode: StatusCodes.Status404NotFound);
+            }
+            catch (Exception err)
+            {
+                _logger.LogInformation("Error: unsuccessful DELETE request. " + err.Message);
+                return StatusCode(statusCode: StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpGet]
+        public async Task<ActionResult> SearchDocument(string language, string code, float threshold, int limit = -1)
```

Comment:
What is the purpose of this endpoint? The GET is supposed to be by ID, not language and "badCode".

Reply:
This is the API that calls the vector search with `badCode`. There's another endpoint `GetDocument(id, language)` that gets an item with the id. Do you have a suggestion for another endpoint name?

Reply:
I should also ask, `GetDocument` takes 2 arguments - `id` and `language`. This is from the assumption that if they know the ID, they will also know the language. `language` is also the partition key for the DB so it queries faster. Is this okay or would you prefer using `id` only for GET?